lib/broccoli/ember-app.js

```diff
@@ -187,7 +187,8 @@ class EmberApp {
     let srcPath = this._resolveLocal('src');
     let srcTree = existsSync(srcPath) ? new WatchedDir(srcPath) : null;
 
-    let appTree = new WatchedDir(this._resolveLocal('app'));
+    let appPath = this._resolveLocal('app');
+    let appTree = existsSync(appPath) ? new WatchedDir(appPath) : null;
```

Comment:
I'm a little worried about the potential side effects of this change since we don't have that many unit tests for the build pipeline yet and the change is not covered by the feature flag